### PR TITLE
Added CI workflow for GitHub Actions

### PR DESCRIPTION
**Pull request by drkameleon**
Since I noticed that you have a setup for Travis and Appveyor, I unified all 3 of them in one config file - and it all works quite neat and organized :)

P.S. I'm not 100% sure if caching (of the dependencies) work flawlessly yet, but I'll look into it. Other than that, all 3 builds (ubuntu-latest, macos-latest and windows-latest) build just fine.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Verify environment
+        run: |
+          stack --numeric-version
+          ghc --version
+          
+      # Build dependencies
+      - name: Build dependencies 
+        run: |
+          cd waspc
+          stack --no-terminal test --only-dependencies
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:84]: You removed --install-ghc, but I would put it back. It does nothing if ghc is already installed, so it does not hurt, and it does make this command a little bit more robust to any changes in the previous actions.

Comment (Martinsos) [inline .github/workflows/ci.yaml:84]: Let's try removing the --no-terminal, maybe Github Actions will handle output nicely.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        if: steps.cache-dependencies.outputs.cache-hit != 'true'
+      
+      # Build Wasp
+      - name: Build Wasp & Run tests
+        run: |
+          cd waspc
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:90]: Inteesting, so we have to do `cd waspc` each time, in each step? In Travis and Appveyor once you do it, you remain there.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,18 @@
 <p align=center>
   <img height="80px" src="https://user-images.githubusercontent.com/1536647/77317442-78625700-6d0b-11ea-9822-0fb21e557e87.png"/>
 </p>
-
-<p align=center> A programming language that understands what a web app is. </p>
+<p align=center>
+  <span>A programming language that understands what a web app is.</span>
```
Comment (Martinsos) [inline README.md:5]: Why this extra span?

File: README.md [hunk 4]
```diff
@@ -90,3 +95,27 @@ You can also:
 # FAQ
 
 Check out https://wasp-lang.dev/#faq.
+
+# License
```
Comment (Martinsos) [inline README.md:99]: We already have License as a separate file (LICENSE), so we can remove this right?

**Review by Martinsos — CHANGES_REQUESTED** [review]
TLDR: Great job! I left comments to take care of. Only thing missing is deployment -> could you add that also?

Ok this is very cool change :)!

I love the idea of everything united in one service/file, and not split in both Travis and Appveyor.
We implemented Travis + Appveyo because I used them before somewhat + it was easy to find other project using them and stealing parts of config.

I just read Github actions docs and ok, they make sense to me and I believe I understand most of the config we have here, and I like it all together, nice job :)!

One thing I am wondering about is, how is stuff happening on Windows, are there any differences we need to think about? For example cache, I don't know if we are caching correctly for the Windows since, I left a comment about that.

The main thing still missing is deployment. I guess here the difference between linux and Win will be most prominent. How should we go about this, could also add this?

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    tags:
+      - v*
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  # Cancel previous actions
+  cleanup:
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:23]: Actually I find this name confusing, I thought at first it is cleaning up some cache or smth. Maybe we could call it "cancel"?

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+          WORKFLOW_ID=$(curl "Authorization: token ${{ github.token }}" https://api.github.com/repos/${{ github.repository }}/actions/workflows/ci.yaml | jq '.id')
+          echo "Workflow id: ${WORKFLOW_ID}"
+          echo "WORKFLOW_ID=${WORKFLOW_ID}" >> $GITHUB_ENV
+
+      - uses: styfle/cancel-workflow-action@0.3.1
+        with:
+          access_token: ${{ github.token }}
+          workflow_id: ${{ env.WORKFLOW_ID }}
+
+  # Main build
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:38]: Redundant comment.

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+          - macos-latest
+          - windows-latest
+    steps:
+      # Checkout
+      - name: Checkout
+        uses: actions/checkout@main
+        with:
+          submodules: recursive
+
+      # Set up cache for dependencies
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:55]: Redundant comment.

File: .github/workflows/ci.yaml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Set up cache
+        id: cache-dependencies
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:57]: What is this id for?

File: .github/workflows/ci.yaml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+      # Setup Haskell
+      - name: Set up environment
+        uses: actions/setup-haskell@v1.1
+        with:
+          ghc-version: 'latest'
+          enable-stack: true
+          stack-version: 'latest'
+
+      # Verify everything's right
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:74]: Redundant comment.

File: .github/workflows/ci.yaml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Verify environment
+        run: |
+          stack --numeric-version
+          ghc --version
+          
+      # Build dependencies
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:80]: Redundant comment.

File: .github/workflows/ci.yaml [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Build dependencies 
+        run: |
+          cd waspc
+          stack --no-terminal test --only-dependencies
+        if: steps.cache-dependencies.outputs.cache-hit != 'true'
+      
+      # Build Wasp
```
Comment (Martinsos) [inline .github/workflows/ci.yaml:87]: Redundant comment.

File: README.md [hunk 12]
```diff
@@ -1,13 +1,18 @@
...
 </p>
-
-<p align=center> A programming language that understands what a web app is. </p>
+<p align=center>
+  <span>A programming language that understands what a web app is.</span>
+</p>
 <br>
+<p align=center>
+  <a href="https://github.com/wasp-lang/wasp/blob/master/LICENSE"><img alt="license" src="https://img.shields.io/github/license/wasp-lang/wasp"></a>
+  <a href="https://github.com/wasp-lang/wasp/search?l=haskell"><img alt="language" src="https://img.shields.io/badge/language-Haskell-purple.svg"></a>
```
Comment (Martinsos) [inline README.md:10]: Ohoo cool we get haskell language badge :D.

Comment (drkameleon) [reply README.md:10]: Hahaha

**Comment by drkameleon** [comment]
@Martinsos Given that I'm an absolute disaster with Git, I deleted my fork and now I cannot find any normal way to make the edits you requested in the original PR.

So, I'm closing this and will make another PR - attempting to address the issues you mentioned.

**Comment by Martinsos** [comment]
> @Martinsos Given that I'm an absolute disaster with Git, I deleted my fork and now I cannot find any normal way to make the edits you requested in the original PR.
> 
> So, I'm closing this and will make another PR - attempting to address the issues you mentioned.

Hah well that is an interesting situation, deleting fork in the middle of PR :D. No probs lets continue there.
